cmake/gotcha_config.h.in

```diff
@@ -0,0 +1,10 @@
+#ifndef GOTCHA_CONFIG_HPP
+#define GOTCHA_CONFIG_HPP
+
+#define GOTCHA_VERSION @GOTCHA_VERSION@
+#define GOTCHA_VERSION_MAJOR (GOTCHA_VERSION / 100000)
+#define GOTCHA_VERSION_MINOR ((GOTCHA_VERSION / 100) % 1000)
+#define GOTCHA_VERSION_PATCH (GOTCHA_VERSION % 100)
+#define GOTCHA_GET_VERSION(MAJOR, MINOR, PATCH) MAJOR * 100000 + MINOR * 100 + PATCH
```

Comment:
and use the tuple here:

```suggestion
#define GOTCHA_GET_VERSION(MAJOR, MINOR, PATCH) (MAJOR * 100000 + MINOR * 100 + PATCH)
#define GOTCHA_VERSION GOTCHA_GET_VERSION @GOTCHA_VERSION@
#define GOTCHA_VERSION_MAJOR (GOTCHA_VERSION / 100000)
#define GOTCHA_VERSION_MINOR ((GOTCHA_VERSION / 100) % 1000)
#define GOTCHA_VERSION_PATCH (GOTCHA_VERSION % 100)
```

it should expand to:

```c
#define GOTCHA_VERSION GOTCHA_GET_VERSION (1, 0, 6)
```

this avoids documenting the `XYYYZZ` enconding